src/inmanta/deploy/tasks.py

```diff
@@ -192,11 +193,14 @@ async def execute(self, task_manager: "scheduler.TaskManager", agent: str, reaso
                         action_id, gid, executor_resource_details, reason, intent.dependencies
                     )
                     # Translate deploy result status to the new deployment result state
-                    match deploy_result.status:
-                        case const.ResourceState.deployed:
+                    match deploy_result.resource_state:
+                        case ContextResourceState.deployed:
                             scheduler_deployment_result = state.DeploymentResult.DEPLOYED
-                        case const.ResourceState.skipped:
+                        case ContextResourceState.skipped:
+                            scheduler_deployment_result = state.DeploymentResult.SKIPPED
+                        case ContextResourceState.skipped_for_dependency:
                             scheduler_deployment_result = state.DeploymentResult.SKIPPED
+                            skipped_for_dependency = True
```

Comment:
I wonder if we should encode this into the DeploymentResult as well?

(purely a question, I don't want to disturb this delicate machinery unduely)

Reply:
I did this in this commit: https://github.com/inmanta/inmanta-core/pull/8383/commits/a957c02cec8ced208a43ebd49a8841a944760dd7

I'm not sure if it changes the behaviour of `report_resource_state` in a way that we do not want to. What do you think?